javalib/src/main/scala/java/util/regex/RegExpImpl.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package java.util.regex
+
+import scala.scalajs.LinkingInfo._
+
+/** A wrapper class to select regex implementation across different platforms.
+ *
+ *  This class provides a common interface for regular expression operations,
+ *  allowing the underlying implementation to be switched at link time.
+ *  For WebAssembly, it uses a Java-based implementation, while for JavaScript
+ *  environments, it delegates to the native `js.RegExp`.
+ */
+private[java] sealed abstract class RegExpImpl {
+  type PatRepr
+  type Repr
+
+  def compile(patternStr: String): PatRepr
+  def exec(pattern: PatRepr, string: String): Repr
+  def matches(r: Repr): scala.Boolean
+  def exists(r: Repr, index: Int): scala.Boolean
```

Comment:
The `scala.` prefixes are not necessary once we're outside the `java.lang` package.